webapp/src/components/sidebar_right/index.tsx

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import Scrollbars from 'react-custom-scrollbars';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {Theme} from 'mattermost-redux/types/preferences';
+import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
+
+import {getYourPrDetails, getReviewDetails} from 'src/actions';
+import {RHSStates} from 'src/constants';
+import {getSidebarData} from 'src/selectors';
+import {Item} from 'src/types/gitlab_items';
+
+import GitlabItems from './gitlab_items';
+
+const AUTO_HIDE_TIMEOUT = 500;
+const AUTO_HIDE_DURATION = 500;
+
+interface Props {
+    username: string;
+    org: string;
+    gitlabURL: string;
+    reviews: Item[];
+    unreads: Item[],
+    yourPrs: Item[],
+    yourAssignments: Item[],
+    rhsState: string,
+    theme: Theme,
+}
+
+export const renderView = (props: Props) => (
+    <div
+        {...props}
+        className='scrollbar--view'
+    />
+);
+
+export const renderThumbHorizontal = (props: Props) => (
+    <div
+        {...props}
+        className='scrollbar--horizontal'
+    />
+);
+
+export const renderThumbVertical = (props: Props) => (
+    <div
+        {...props}
+        className='scrollbar--vertical'
+    />
+);
+
+function SidebarRight({theme}: {theme: Theme}) {
+    const sidebarData = useSelector(getSidebarData);
+    const {username, yourAssignments, org, unreads, gitlabURL, rhsState, reviews, yourPrs} = sidebarData;
+
+    const dispatch = useDispatch();
+
+    // Get the details for PRs and reviews on the first render
+    useEffect(() => {
+        if (yourPrs && rhsState === RHSStates.PRS) {
+            dispatch(getYourPrDetails(yourPrs));
+        }
+
+        if (reviews && rhsState === RHSStates.REVIEWS) {
+            dispatch(getReviewDetails(reviews));
+        }
+    }, []);
+
+    useEffect(() => {
+        if (RHSStates.PRS === rhsState) {
```

Comment:
I think this reads better

```suggestion
        if (rhsState === RHSStates.PRS) {
```